### PR TITLE
Fix/https agent is not applied correctly

### PR DESCRIPTION
**Pull request by jkoenig134**

### CODE REVIEW

File: packages/transport/src/core/backbone/RESTClient.ts [hunk 1]
```diff
@@ -81,17 +82,24 @@ export class RESTClient {
             defaults.headers = _.defaultsDeep({}, defaults.headers, this.config.platformAdditionalHeaders);
         }
 
+        if (config.httpsAgent) {
+            defaults.httpsAgent = config.httpsAgent;
+        } else {
+            try {
+                // eslint-disable-next-line @typescript-eslint/no-require-imports
+                const httpsAgent = require("https")?.Agent;
```
Comment (sebbi08) [inline packages/transport/src/core/backbone/RESTClient.ts:90]: Why is this required here when it is imported at the start of the file ?

File: packages/transport/src/core/backbone/RESTClient.ts [hunk 2]
```diff
@@ -81,17 +82,24 @@ export class RESTClient {
...
+
+                if (httpsAgent) defaults.httpsAgent = new httpsAgent(this.config.httpsAgentOptions);
+            } catch (e) {
+                // ignore
+            }
+        }
+
         try {
             // eslint-disable-next-line @typescript-eslint/no-require-imports
             const agent = require("http")?.Agent;
```
Comment (sebbi08) [inline packages/transport/src/core/backbone/RESTClient.ts:100]: Same why is this required and not imported ?

File: packages/transport/src/core/backbone/RESTClient.ts (hunk 1 above)
Comment (jkoenig134) [reply packages/transport/src/core/backbone/RESTClient.ts:90]: it's not available in browsers
